circuit/src/verifier/aggregator/mod.rs

```diff
@@ -132,58 +120,14 @@ impl AggregatorConfig {
 	}
 }
 
-impl Circuit<Fr> for Aggregator {
+impl Chipset<Fr> for AggregatorChipset {
 	type Config = AggregatorConfig;
-	type FloorPlanner = SimpleFloorPlanner;
-
-	/// Returns a copy of this circuit with no witness values
-	fn without_witnesses(&self) -> Self {
-		Self {
-			svk: self.svk,
-			snarks: self.snarks.iter().map(UnassignedSnark::without_witness).collect(),
-			as_proof: None,
-		}
-	}
-
-	/// Configure the circuit.
-	fn configure(meta: &mut ConstraintSystem<Fr>) -> Self::Config {
-		let common = CommonConfig::new(meta);
-		let main_selector = MainChip::configure(&common, meta);
-		let main = MainConfig::new(main_selector);
-
-		let full_round_selector = FullRoundHasher::configure(&common, meta);
-		let partial_round_selector = PartialRoundHasher::configure(&common, meta);
-		let poseidon = PoseidonConfig::new(full_round_selector, partial_round_selector);
-
-		let absorb_selector = AbsorbChip::<Fr, WIDTH>::configure(&common, meta);
-		let poseidon_sponge = PoseidonSpongeConfig::new(poseidon, absorb_selector);
-
-		let bits2num = Bits2NumChip::configure(&common, meta);
-
-		let int_red =
-			IntegerReduceChip::<Fq, Fr, NUM_LIMBS, NUM_BITS, Bn256_4_68>::configure(&common, meta);
-		let int_add =
-			IntegerAddChip::<Fq, Fr, NUM_LIMBS, NUM_BITS, Bn256_4_68>::configure(&common, meta);
-		let int_sub =
-			IntegerSubChip::<Fq, Fr, NUM_LIMBS, NUM_BITS, Bn256_4_68>::configure(&common, meta);
-		let int_mul =
-			IntegerMulChip::<Fq, Fr, NUM_LIMBS, NUM_BITS, Bn256_4_68>::configure(&common, meta);
-		let int_div =
-			IntegerDivChip::<Fq, Fr, NUM_LIMBS, NUM_BITS, Bn256_4_68>::configure(&common, meta);
-
-		let ladder = EccUnreducedLadderConfig::new(int_add, int_sub, int_mul, int_div);
-		let add = EccAddConfig::new(int_red, int_sub, int_mul, int_div);
-		let double = EccDoubleConfig::new(int_red, int_add, int_sub, int_mul, int_div);
-		let table_select = EccTableSelectConfig::new(main.clone());
-		let ecc_mul_scalar = EccMulConfig::new(ladder, add, double, table_select, bits2num);
-
-		AggregatorConfig { common, main, poseidon_sponge, ecc_mul_scalar }
-	}
+	type Output = Vec<AssignedCell<Fr, Fr>>;
 
 	/// Synthesize the circuit.
 	fn synthesize(
-		&self, config: Self::Config, mut layouter: impl Layouter<Fr>,
-	) -> Result<(), Error> {
+		self, _common: &CommonConfig, config: &Self::Config, mut layouter: impl Layouter<Fr>,
```

Comment:
We should use this common instead of keeping it in `AggregatorConfig`

Reply:
Hmm, you mean that we should remove the `common` from `AggregatorConfig`, and use the above one?

Reply:
Yes, exactly